### PR TITLE
deprecate aligned_storage

### LINKED ISSUE #1253: std::aligned_storage is deprecated in C++23 and should not be used anymore
https://eel.is/c++draft/depr.meta.types

### PR DESCRIPTION
**Pull request by blonded04**
### Description 
Removed deprecated `aligned_storage_t` from tbb

Fixes #1253

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [x] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: include/oneapi/tbb/detail/_concurrent_unordered_base.h [hunk 1]
```diff
@@ -179,8 +179,7 @@ class value_node : public list_node<SokeyType>
     }
 
 private:
-    using aligned_storage_type = typename std::aligned_storage<sizeof(value_type)>::type;
-    aligned_storage_type my_value;
+    alignas(value_type) char my_value[sizeof(value_type)];
```
Comment (blonded04) [inline include/oneapi/tbb/detail/_concurrent_unordered_base.h:182]: is `alignas(max_nfs_size)` missing here?

**Comment by blonded04** [comment]
Hello @kboyarinov ! Can you please review it?

**Comment by blonded04** [comment]
@kboyarinov Any chance it's merged?

**Review by kboyarinov — COMMENTED** [review]
@blonded04,

Sorry for taking so long to review this. I am OK with the idea of the change.

But the problem is that for the previous implementation, the second argument for `std::aligned_storage` was missed and hence the `my_value` field alignment can change if the change would be applied that results in a breaking change (consider the use-case when the container is passed between the border of the translation units built with different oneTBB versions). 
To protect the user from this, we need to update the implementation version for all of the concurrent unordered containers by changing the internal namespace number. Currently it is `d1`. For this change, we need to update it to `d2`. Could you please do it as part of this PR to make it possible to merge it?

**Comment by blonded04** [comment]
> @blonded04,
> 
> Sorry for taking so long to review this. I am OK with the idea of the change.
> 
> But the problem is that for the previous implementation, the second argument for `std::aligned_storage` was missed and hence the `my_value` field alignment can change if the change would be applied that results in a breaking change (consider the use-case when the container is passed between the border of the translation units built with different oneTBB versions). To protect the user from this, we need to update the implementation version for all of the concurrent unordered containers by changing the internal namespace number. Currently it is `d1`. For this change, we need to update it to `d2`. Could you please do it as part of this PR to make it possible to merge it?

Thanks for review. I'll finish this PR on this weekend.

**Comment by blonded04** [comment]
Okay, thank you, I'll do it on the closest weekend!

**Comment by blonded04** [comment]
> > @blonded04,
> > Sorry for taking so long to review this. I am OK with the idea of the change.
> > But the problem is that for the previous implementation, the second argument for `std::aligned_storage` was missed and hence the `my_value` field alignment can change if the change would be applied that results in a breaking change (consider the use-case when the container is passed between the border of the translation units built with different oneTBB versions). To protect the user from this, we need to update the implementation version for all of the concurrent unordered containers by changing the internal namespace number. Currently it is `d1`. For this change, we need to update it to `d2`. Could you please do it as part of this PR to make it possible to merge it?
> 
> Thanks for review. I'll finish this PR on this weekend.

next weekend* sorry

**Pull request closed without merging**